packages/node_modules/@ciscospark/internal-plugin-board/src/board.js

```diff
@@ -147,10 +149,31 @@ const Board = SparkPlugin.extend({
       aclLinkOperation: `DELETE`
     };
 
+    let promise = Promise.resolve();
+    if (options.preventDeleteActiveChannel) {
+      promise = this.lockChannelForDeletion(channel);
+    }
+
+    return promise
+      .then(() => this.spark.request({
+        method: `PUT`,
+        uri: `${channel.aclUrl}/links`,
+        body
+      }))
+      .then((res) => res.body);
+  },
+
+  /**
+   * Locks and marks a channel for deletion
+   * If a channel is being used, it will return 409 - Conflict
+   * @memberof Board.BoardService
+   * @param  {Board~Channel} channel
+   * @returns {Promise}
+   */
+  lockChannelForDeletion(channel) {
     return this.spark.request({
-      method: `PUT`,
-      uri: `${channel.aclUrl}/links`,
-      body
+      method: `POST`,
+      uri: `${channel.channelUrl}/lock?intent=delete`
```

Comment:
is this a case where qs can actually be used? (unlike my confusion last time)

Reply:
does the channel have a links array? is there a lock link that can dropped directly into the url instead of synthesizing it?

Reply:
Channel doesn't have links array. Let me put in a request for that?

Reply:
nah, it's fine, i just saw that you stopped using  `${channel.aclUrl}/links`, so i wanted to make sure we didn't have something like that already.

Reply:
oh it's just weird diff issues... I didn't remove that part of the code. It belonged in a different function